File: TOMIMSV4/Entity/Mantenimientos/Bodega/Bodega_Ubicacion/clsBeBodega_ubicacion.vb
<Serializable>
Public Class clsBeBodega_ubicacion
    Implements ICloneable
    Public Property IdUbicacion() As Integer = 0
    Public Property IdTramo() As Integer = 0
    Public Property IdSector() As Integer = 0
    Public Property IdArea() As Integer = 0
    Public Property IdBodega() As Integer = 0
    Public Property Descripcion() As String = ""
    Public Property Ancho() As Double = 0.0
    Public Property Largo() As Double = 0.0
    Public Property Alto() As Double = 0.0
    Public Property Nivel() As Integer = 0
    Public Property Indice_x() As Integer = 0
    Public Property IdIndiceRotacion() As Integer = 0
    Public Property IdTipoRotacion() As Integer = 0
    Public Property Sistema() As Boolean = False
    Public Property Codigo_barra() As String = ""
    Public Property Codigo_barra2() As String = ""
    Public Property User_agr() As String = ""
    Public Property Fec_agr() As Date = Date.Now
    Public Property User_mod() As String = ""
    Public Property Fec_mod() As Date = Date.Now
    Public Property Dañado() As Boolean = False
    Public Property Activo() As Boolean = False
    Public Property Bloqueada() As Boolean = False
    Public Property Acepta_pallet() As Boolean = False
    Public Property Ubicacion_picking() As Boolean = False
    Public Property Ubicacion_recepcion() As Boolean = False
    Public Property Ubicacion_despacho() As Boolean = False
    Public Property Ubicacion_merma() As Boolean = False
    Public Property Ubicacion_Virtual As Boolean = False
    Public Property Margen_izquierdo() As Double = 0.0
    Public Property Margen_derecho() As Double = 0.0
    Public Property Margen_superior() As Double = 0.0
    Public Property Margen_inferior() As Double = 0.0
    Public Property Orientacion_pos() As String = ""
    'GT 30082021: CAMBIO DE VALOR
    'GT 20211227 Lo cambié nuevamente a Boolean
    Public Property ubicacion_ne() As Boolean = False
    ''' <summary>
    ''' Se calcula en demanda, se utiliza en el procso de ubicación para determinar cuanto espacio (porcentualmente) tiene disponible la ubicación.
    ''' </summary>
    ''' <returns></returns>
    Public Property Disponibilidad_Ubicacion As Double = 0
    Public Property Posicion_X() As Double = 0
    Public Property Posicion_Y() As Double = 0
    ''' <summary>
    '''#EJC20240609: Ubicación tipo muelle.
    ''' </summary>
    ''' <returns></returns>
    Public Property Ubicacion_muelle() As Boolean = False
    Sub New()
    End Sub
    Sub New(ByRef IdUbicacion As Integer, ByVal IdTramo As Integer, ByVal descripcion As String, ByVal ancho As Double, ByVal largo As Double, ByVal alto As Double, ByVal nivel As Integer, ByVal indice_x As Integer, ByVal IdIndiceRotacion As Integer, ByVal IdTipoRotacion As Integer, ByVal sistema As Boolean, ByVal codigo_barra As String, ByVal codigo_barra2 As String, ByVal user_agr As String, ByVal fec_agr As Date, ByVal user_mod As String, ByVal fec_mod As Date, ByVal dañado As Boolean, ByVal activo As Boolean, ByVal bloqueada As Boolean, ByVal acepta_pallet As Boolean, ByVal ubicacion_picking As Boolean, ByVal ubicacion_recepcion As Boolean, ByVal ubicacion_despacho As Boolean, ByVal ubicacion_merma As Boolean, ByVal margen_izquierdo As Double, ByVal margen_derecho As Double, ByVal margen_superior As Double, ByVal margen_inferior As Double, ByVal orientacion_pos As String, ByVal ubicacion_ne As Boolean)
        Me.IdUbicacion = IdUbicacion
        Me.IdTramo = IdTramo
        Me.Descripcion = descripcion
        Me.Ancho = ancho
        Me.Largo = largo
        Me.Alto = alto
        Me.Nivel = nivel
        Me.Indice_x = indice_x
        Me.IdIndiceRotacion = IdIndiceRotacion
        Me.IdTipoRotacion = IdTipoRotacion
        Me.Sistema = sistema
        Me.Codigo_barra = codigo_barra
        Me.Codigo_barra2 = codigo_barra2
        Me.User_agr = user_agr
        Me.Fec_agr = fec_agr
        Me.User_mod = user_mod
        Me.Fec_mod = fec_mod
        Me.Dañado = dañado
        Me.Activo = activo
        Me.Bloqueada = bloqueada
        Me.Acepta_pallet = acepta_pallet
        Me.Ubicacion_picking = ubicacion_picking
        Me.Ubicacion_recepcion = ubicacion_recepcion
        Me.Ubicacion_despacho = ubicacion_despacho
        Me.Ubicacion_merma = ubicacion_merma
        Me.Margen_izquierdo = margen_izquierdo
        Me.Margen_derecho = margen_derecho
        Me.Margen_superior = margen_superior
        Me.Margen_inferior = margen_inferior
        Me.Orientacion_pos = orientacion_pos
        Me.ubicacion_ne = ubicacion_ne
    End Sub
    Public Function Clone() As Object Implements System.ICloneable.Clone
        Return MyBase.MemberwiseClone()
    End Function

End Class

File: WMS.EntityCore/Datos_Maestros/clsBeBodega_ubicacion.cs
using System;
using System.ComponentModel;
using System.ComponentModel.DataAnnotations.Schema;

namespace WMSWebAPI.Be
{
    public class clsBeBodega_ubicacion : ICloneable
    {
        [Column("IdUbicacion")]
        [DisplayName("IdUbicacion")]
        public int IdUbicacion { get; set; } = 0;

        [Column("IdTramo")]
        [DisplayName("IdTramo")]
        public int IdTramo { get; set; } = 0;

        [Column("descripcion")]
        [DisplayName("descripcion")]
        public string Descripcion { get; set; } = "";

        [Column("ancho")]
        [DisplayName("ancho")]
        public double Ancho { get; set; } = 0;

        [Column("largo")]
        [DisplayName("largo")]
        public double Largo { get; set; } = 0;

        [Column("alto")]
        [DisplayName("alto")]
        public double Alto { get; set; } = 0;

        [Column("nivel")]
        [DisplayName("nivel")]
        public int Nivel { get; set; } = 0;

        [Column("indice_x")]
        [DisplayName("indice_x")]
        public int Indice_x { get; set; } = 0;

        [Column("IdIndiceRotacion")]
        [DisplayName("IdIndiceRotacion")]
        public int IdIndiceRotacion { get; set; } = 0;

        [Column("IdTipoRotacion")]
        [DisplayName("IdTipoRotacion")]
        public int IdTipoRotacion { get; set; } = 0;

        [Column("sistema")]
        [DisplayName("sistema")]
        public bool Sistema { get; set; } = false;

        [Column("codigo_barra")]
        [DisplayName("codigo_barra")]
        public string Codigo_barra { get; set; } = "";

        [Column("codigo_barra2")]
        [DisplayName("codigo_barra2")]
        public string Codigo_barra2 { get; set; } = "";

        [Column("user_agr")]
        [DisplayName("user_agr")]
        public string User_agr { get; set; } = "";

        [Column("fec_agr")]
        [DisplayName("fec_agr")]
        public DateTime Fec_agr { get; set; } = DateTime.Now;

        [Column("user_mod")]
        [DisplayName("user_mod")]
        public string User_mod { get; set; } = "";

        [Column("fec_mod")]
        [DisplayName("fec_mod")]
        public DateTime Fec_mod { get; set; } = DateTime.Now;

        [Column("dañado")]
        [DisplayName("dañado")]
        public bool Dañado { get; set; } = false;

        [Column("activo")]
        [DisplayName("activo")]
        public bool Activo { get; set; } = false;

        [Column("bloqueada")]
        [DisplayName("bloqueada")]
        public bool Bloqueada { get; set; } = false;

        [Column("acepta_pallet")]
        [DisplayName("acepta_pallet")]
        public bool Acepta_pallet { get; set; } = false;

        [Column("ubicacion_picking")]
        [DisplayName("ubicacion_picking")]
        public bool Ubicacion_picking { get; set; } = false;

        [Column("ubicacion_recepcion")]
        [DisplayName("ubicacion_recepcion")]
        public bool Ubicacion_recepcion { get; set; } = false;

        [Column("ubicacion_despacho")]
        [DisplayName("ubicacion_despacho")]
        public bool Ubicacion_despacho { get; set; } = false;

        [Column("ubicacion_merma")]
        [DisplayName("ubicacion_merma")]
        public bool Ubicacion_merma { get; set; } = false;

        [Column("margen_izquierdo")]
        [DisplayName("margen_izquierdo")]
        public double Margen_izquierdo { get; set; } = 0;

        [Column("margen_derecho")]
        [DisplayName("margen_derecho")]
        public double Margen_derecho { get; set; } = 0;

        [Column("margen_superior")]
        [DisplayName("margen_superior")]
        public double Margen_superior { get; set; } = 0;

        [Column("margen_inferior")]
        [DisplayName("margen_inferior")]
        public double Margen_inferior { get; set; } = 0;

        [Column("orientacion_pos")]
        [DisplayName("orientacion_pos")]
        public string Orientacion_pos { get; set; } = "";

        [Column("ubicacion_virtual")]
        [DisplayName("ubicacion_virtual")]
        public bool Ubicacion_virtual { get; set; } = false;

        [Column("ubicacion_ne")]
        [DisplayName("ubicacion_ne")]
        public bool Ubicacion_ne { get; set; } = false;

        [Column("IdBodega")]
        [DisplayName("IdBodega")]
        public int IdBodega { get; set; } = 0;

        [Column("IdArea")]
        [DisplayName("IdArea")]
        public int IdArea { get; set; } = 0;

        [Column("IdSector")]
        [DisplayName("IdSector")]
        public int IdSector { get; set; } = 0;

        [Column("posicion_x")]
        [DisplayName("posicion_x")]
        public double Posicion_x { get; set; } = 0;

        [Column("posicion_y")]
        [DisplayName("posicion_y")]
        public double Posicion_y { get; set; } = 0;

        [Column("ubicacion_muelle")]
        [DisplayName("ubicacion_muelle")]
        public bool Ubicacion_muelle { get; set; } = false;

        public clsBeBodega_ubicacion() { }

        public object Clone()
        {
            return this.MemberwiseClone();
        }
    }
}
